### ISSUE TITLE
error: invalid value for type 'layout'. Expected a reference.

### ISSUE THREAD

**Issue by mastercho** [shroudedcode/apk-mitm#1]
Here some logs

```    Signing patched APK file

   Failed!  An error occurred:

 Error: Command failed: java -jar C:\Users\Mastercho\AppData\Roaming\npm\node_modules\apk-mitm\jar\apktool.jar build C:\Users\MASTER~1\AppData\Local\Temp\0034fb560faa867748c450aa2405ed3e\decode --output C:\Users\MASTER~1\AppData\Local\Temp\0034fb560faa867748c450aa2405ed3e\unsigned.apk --use-aapt2
W: C:\Users\MASTER~1\AppData\Local\Temp\0034fb560faa867748c450aa2405ed3e\decode\res\values\layouts.xml:3: error: invalid value for type 'layout'. Expected a reference.
```

Can be something wrong with java? Java version is 8u231

**Comment by shroudedcode** [comment]
> Can be something wrong with java?

That's very unlikely. It's either an issue with [Apktool](https://github.com/iBotPeaches/Apktool) or there's actually something wrong with the `res\values\layouts.xml` file in question. `apk-mitm` does not delete its temporary directories, so I'd recommend taking a look at it (specifically line 3).

**Comment by mastercho** [comment]
Tried couple apks same error on everything so might be Apktool, after doing some research in StackOverflow, i can see that is an existing issue in Apktool

**Comment by shroudedcode** [comment]
@mastercho Could you try patching that APK file again with the latest version ([v0.4.0](https://github.com/shroudedcode/apk-mitm/releases/tag/v0.4.0)) of `apk-mitm`? I've made [some changes](https://github.com/shroudedcode/apk-mitm/issues/5#issuecomment-546745176) to the decoding step, so it might work now.

**Comment by mastercho** [comment]
did you push updates to npm package, as `npm update -g apk-mitm` not finding any updates

**Comment by mastercho** [comment]
I did update manually and tried 3 APKs Instagram, TikTok and AutoZone first 2 gives same error. AutoZone throws that error:

```
Error: Command failed: java -jar C:\Users\Mastercho\AppData\Roaming\npm\node_modules\apk-mitm\jar\apktool.jar decode C:\Users\Mastercho\Downloads\Autozone.apk --output C:\Users\MASTER~1\AppData\Local\Temp\cc3a2ff45e327c9538175e43b7774077\decode
S: Could not decode file, replacing by FALSE value: drawable-hdpi-v4/abc_menu_hardkey_panel_mtrl_mult.9.png
Exception in thread "main" brut.androlib.AndrolibException: Could not decode raw stream
```

I though apktool is the issue but APK Studio uses same version 2.4.0  and there no issues like this

**Comment by shroudedcode** [comment]
I'm closing this issue because:

- I can patch Instagram's latest APK without issues when using the latest `apk-mitm` version
- the issue you run into with TikTok can only by fixed in Apktool *(upvote [#1990](https://github.com/iBotPeaches/Apktool/issues/1990), [#2099](https://github.com/iBotPeaches/Apktool/issues/2099), and [#2249](https://github.com/iBotPeaches/Apktool/issues/2249))*
